cty/json/type_implied.go

```diff
@@ -150,10 +150,12 @@ func impliedTupleType(dec *json.Decoder) (cty.Type, error) {
 		}
 
 		if ttok, ok := tok.(json.Delim); ok {
-			if rune(ttok) != ']' {
+			if rune(ttok) != '{' && rune(ttok) != ']' {
```

Comment:
It looks like `impledTypeForTok` already handles the case of an unsupported delimiter, so I wonder if we could get the same effect here with just the `if rune(ttok) == ']'` branch below... what do you think?

Reply:
Yeah I did that initially, but thought to change as little as possible (not knowing the background of why this was here as well). Will update the PR.

Reply:
Updated... This actually also supports lists in lists, which would still be missed by my previous fix.